### PR TITLE
clarify deploy-full-rp-service-in-dev.md guide

### PR DESCRIPTION
**Pull request by mbukatov**
### Which issue this PR addresses:

None.

### What this PR does / why we need it:

Improves "Deploy an Entire RP Development Service" guide so that one can avoid problems caused by misunderstanding of the procedure.

### Test plan for issue:

Changes are based on my experience with this procedure. No further testing expected beyond review.

### CODE REVIEW

**Comment by mbukatov** [comment]
Publishing as a draft for visibility, I still need to polish this a bit. Feedback welcome.

File: docs/deploy-full-rp-service-in-dev.md [hunk 1]
```diff
@@ -3,31 +3,56 @@
 ## Prerequisites
 
 1. Your development environment is prepared according to the steps outlined in [Prepare Your Dev Environment](./prepare-your-dev-environment.md)
-
+2. During the deployment, don't edit files in your ARO-RP repository so that
+   `git status` reports a clean working tree. Otherwise aro container image
+   will have `-dirty` suffix, which can cause problems esp. when the repository
+   becomes dirty during the procedure.
```
Comment (yehornaumenko) [inline docs/deploy-full-rp-service-in-dev.md:9]: I guess we need to prevent this programmatically in this scenario because it can't be wanted. Unless somebody wanna deploy an RP using some not committed changes.
I would add some checks or warning messages to the scripts we use or _make_'s targets.
But we can do it in another PR. Does it make sense?

Comment (jaitaiwan) [reply docs/deploy-full-rp-service-in-dev.md:9]: What sort of problems does the -dirty suffix cause? Is it only an issue when deploying a shared rp or is it an issue for local dev as well?

Comment (SrinivasAtmakuri) [reply docs/deploy-full-rp-service-in-dev.md:9]: One solution would be to set COMMIT/TAG = latest if RP_MODE == development, this will make reruns or testing a lot easier.

Comment (cadenmarchese) [reply docs/deploy-full-rp-service-in-dev.md:9]: To me it's more of a recommendation. You can definitely deploy with the dirty tag if you want (I do it and sometimes it's an accident but it still works), but you will have the potential issues mentioned above.

Should we just change the wording to "it's recommended to..." and add the content here to the document? https://github.com/Azure/ARO-RP/pull/2964/files#r1238448125

Comment (mbukatov) [reply docs/deploy-full-rp-service-in-dev.md]: I updated the text as suggested by @cadenmarchese .

**Comment by mbukatov** [comment]
The changes are ready for review. I addressed the feedback about the dirty image tag, and tweaked the text about container registry used for `make publish-image*` targets.

**Review by tiguelu — DISMISSED** [review]
LGTM

**Review by cadenmarchese — DISMISSED** [review]

**Comment by mbukatov** [comment]
Rebased on top of current master (14e3c5a88) to unblock the github checks.

**Review by tiguelu — APPROVED** [review]
